From 260465fff026085a5ea6a11811c609c7b203a2ae Mon Sep 17 00:00:00 2001
From: Lluis Campos <lluis.campos@northern.tech>
Date: Wed, 11 Mar 2020 15:32:22 +0100
Subject: [PATCH 1/1] MEN-3277: Fix check-update and send-inventory options on
 deb install

When installing mender client through deb package, the systemd service
is renamed to mender-client, so these two CLI options were failing in
sending the signal to the daemon.

Changelog: Title

Signed-off-by: Lluis Campos <lluis.campos@northern.tech>
(cherry picked from commit f49fbf4e0785cb7c3c027bdfab6b9f86130aa534)
---
 cli/cli.go | 24 ++++++++++++++++++++----
 1 file changed, 20 insertions(+), 4 deletions(-)

diff --git a/src/github.com/mendersoftware/mender/cli/cli.go b/src/github.com/mendersoftware/mender/cli/cli.go
index 00e7fa7..b25a55e 100644
--- a/src/github.com/mendersoftware/mender/cli/cli.go
+++ b/src/github.com/mendersoftware/mender/cli/cli.go
@@ -142,11 +142,19 @@ func SetupCLI(args []string) error {
 			Name:  "check-update",
 			Usage: "Force update check.",
 			Action: func(_ *cli.Context) error {
-				return updateCheck(
+				err := updateCheck(
 					exec.Command("kill", "-USR1"),
 					exec.Command("systemctl",
 						"show", "-p",
-						"MainPID", "mender-client"))
+						"MainPID", "mender"))
+				if err != nil {
+					err = updateCheck(
+						exec.Command("kill", "-USR1"),
+						exec.Command("systemctl",
+							"show", "-p",
+							"MainPID", "mender-client"))
+				}
+				return err
 			},
 		},
 		{
@@ -183,11 +191,19 @@ func SetupCLI(args []string) error {
 			Name:  "send-inventory",
 			Usage: "Force inventory update.",
 			Action: func(_ *cli.Context) error {
-				return updateCheck(
+				err := updateCheck(
 					exec.Command("kill", "-USR2"),
 					exec.Command("systemctl",
 						"show", "-p",
-						"MainPID", "mender-client"))
+						"MainPID", "mender"))
+				if err != nil {
+					err = updateCheck(
+						exec.Command("kill", "-USR2"),
+						exec.Command("systemctl",
+							"show", "-p",
+							"MainPID", "mender-client"))
+				}
+				return err
 			},
 		},
 		{
-- 
2.17.1

